src/components/NcRichText/NcReferenceWidget.vue

```diff
@@ -62,29 +62,25 @@ export default {
 	},
 
 	setup() {
-		const compact = ref(3)
+		const width = ref(700)
 		const isVisible = ref(false)
 		// This is the widget root node
 		const widgetRoot = ref()
 
 		useIntersectionObserver(widgetRoot, (entries) => {
-			isVisible.value = entries[0]?.isIntersecting ?? false
+			window.requestAnimationFrame(() => {
+				isVisible.value = entries[0]?.isIntersecting ?? false
+			})
 		})
 
 		useResizeObserver(widgetRoot, (entries) => {
-			if (entries[0].contentRect.width < 450) {
-				compact.value = 0
-			} else if (entries[0].contentRect.width < 550) {
-				compact.value = 1
-			} else if (entries[0].contentRect.width < 650) {
-				compact.value = 2
-			} else {
-				compact.value = 3
-			}
+			window.requestAnimationFrame(() => {
+				width.value = entries[0].contentRect.width
+			})
```

Comment:
If that is only for Cypress, I'd add a comment. Otherwise it is very non-obious for my why there is a need for RAF

Reply:
Frankly I am not sure. In another case I looked into in text I was seeing a
`"ResizeObserver loop completed with undelivered notifications"`
error right before the crash.

I did look into that some and it seems to indicate that there is a problem with the callback. Some cause can be an infinite loop where the resize triggers the observer which triggers a callback that triggers another resize.

https://blog.flowpoint.ai/post/v2/solved-resizeobserver-loop-completed-with-undelivered-notifications/

We could also use a debounce but that seemed even more involved.

Reply:
A `nextTick` also does the trick and seems more light weight.